// @angular/forms/forms.ngfactory.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
define(["require", "exports", "@angular/core", "@angular/forms"], function (require, exports, import0, import1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    var FormsModuleInjector = (function (_super) {
        __extends(FormsModuleInjector, _super);
        function FormsModuleInjector(parent) {
            return _super.call(this, parent, [], []) || this;
        }
        Object.defineProperty(FormsModuleInjector.prototype, "_\u0275i_2", {
            get: function () {
                if ((this.__ɵi_2 == null)) {
                    (this.__ɵi_2 = new import1.ɵi());
                }
                return this.__ɵi_2;
            },
            enumerable: true,
            configurable: true
        });
        FormsModuleInjector.prototype.createInternal = function () {
            this._ɵba_0 = new import1.ɵba();
            this._FormsModule_1 = new import1.FormsModule();
            return this._FormsModule_1;
        };
        FormsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
            if ((token === import1.ɵba)) {
                return this._ɵba_0;
            }
            if ((token === import1.FormsModule)) {
                return this._FormsModule_1;
            }
            if ((token === import1.ɵi)) {
                return this._ɵi_2;
            }
            return notFoundResult;
        };
        FormsModuleInjector.prototype.destroyInternal = function () {
        };
        return FormsModuleInjector;
    }(import0.ɵNgModuleInjector));
    exports.FormsModuleNgFactory = new import0.NgModuleFactory(FormsModuleInjector, import1.FormsModule);
    var ɵbaInjector = (function (_super) {
        __extends(ɵbaInjector, _super);
        function ɵbaInjector(parent) {
            return _super.call(this, parent, [], []) || this;
        }
        ɵbaInjector.prototype.createInternal = function () {
            this._ɵba_0 = new import1.ɵba();
            return this._ɵba_0;
        };
        ɵbaInjector.prototype.getInternal = function (token, notFoundResult) {
            if ((token === import1.ɵba)) {
                return this._ɵba_0;
            }
            return notFoundResult;
        };
        ɵbaInjector.prototype.destroyInternal = function () {
        };
        return ɵbaInjector;
    }(import0.ɵNgModuleInjector));
    exports.ɵbaNgFactory = new import0.NgModuleFactory(ɵbaInjector, import1.ɵba);
    var ReactiveFormsModuleInjector = (function (_super) {
        __extends(ReactiveFormsModuleInjector, _super);
        function ReactiveFormsModuleInjector(parent) {
            return _super.call(this, parent, [], []) || this;
        }
        Object.defineProperty(ReactiveFormsModuleInjector.prototype, "_FormBuilder_2", {
            get: function () {
                if ((this.__FormBuilder_2 == null)) {
                    (this.__FormBuilder_2 = new import1.FormBuilder());
                }
                return this.__FormBuilder_2;
            },
            enumerable: true,
            configurable: true
        });
        Object.defineProperty(ReactiveFormsModuleInjector.prototype, "_\u0275i_3", {
            get: function () {
                if ((this.__ɵi_3 == null)) {
                    (this.__ɵi_3 = new import1.ɵi());
                }
                return this.__ɵi_3;
            },
            enumerable: true,
            configurable: true
        });
        ReactiveFormsModuleInjector.prototype.createInternal = function () {
            this._ɵba_0 = new import1.ɵba();
            this._ReactiveFormsModule_1 = new import1.ReactiveFormsModule();
            return this._ReactiveFormsModule_1;
        };
        ReactiveFormsModuleInjector.prototype.getInternal = function (token, notFoundResult) {
            if ((token === import1.ɵba)) {
                return this._ɵba_0;
            }
            if ((token === import1.ReactiveFormsModule)) {
                return this._ReactiveFormsModule_1;
            }
            if ((token === import1.FormBuilder)) {
                return this._FormBuilder_2;
            }
            if ((token === import1.ɵi)) {
                return this._ɵi_3;
            }
            return notFoundResult;
        };
        ReactiveFormsModuleInjector.prototype.destroyInternal = function () {
        };
        return ReactiveFormsModuleInjector;
    }(import0.ɵNgModuleInjector));
    exports.ReactiveFormsModuleNgFactory = new import0.NgModuleFactory(ReactiveFormsModuleInjector, import1.ReactiveFormsModule);
});
